addons/account_asset/account_asset_invoice.py

```diff
@@ -55,11 +55,12 @@ def asset_create(self, cr, uid, lines, context=None):
         asset_obj.write(cr, SUPERUSER_ID, asset_ids, {'active': False})
         for line in lines:
             if line.asset_category_id:
+                sign = -1 if line.invoice_id.type == "in_refund" else 1
```

Comment:
I suppose this is also valable for deffered revenues, so you should do

-1 if line.invoice_id.type in ("in_refund", 'out_refund') else 1